### PR TITLE
Split out Instrumentation logic from WidgetTester

### PR DESCRIPTION
**Pull request by eseidelGoogle**
This will allow writing tests/benchmark which want to use
the engine's default beginFrame and normal passage of time.

@Hixie

### CODE REVIEW

File: packages/flutter_test/lib/src/instrumentation.dart [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+typedef Point SizeToPointFunction(Size size);
+
+/// Helper class for flutter tests providing event dispatch.
+///
+/// This class provides hooks for accessing the rendering tree and dispatching
+/// fake tap/drag/etc. events.
+class Instrumentation {
+    Instrumentation()
+      : binding = WidgetFlutterBinding.ensureInitialized()
+    {}
```
Comment (Hixie) [inline packages/flutter_test/lib/src/instrumentation.dart]: empty constructors don't need braces. replace with semicolon.

File: packages/flutter_test/lib/src/instrumentation.dart [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import 'test_pointer.dart';
+
+typedef Point SizeToPointFunction(Size size);
+
+/// Helper class for flutter tests providing event dispatch.
+///
+/// This class provides hooks for accessing the rendering tree and dispatching
+/// fake tap/drag/etc. events.
+class Instrumentation {
+    Instrumentation()
```
Comment (Hixie) [inline packages/flutter_test/lib/src/instrumentation.dart]: two character indent in dart

File: packages/flutter_test/lib/src/instrumentation.dart [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        ContainerLayer root = layer;
+        Layer child = root.firstChild;
+        while (child != null) {
+          result.addAll(_layers(child));
+          child = child.nextSibling;
+        }
+      }
+      return result;
+    }
+    List<Layer> get layers => _layers(binding.renderView.layer);
```
Comment (Hixie) [inline packages/flutter_test/lib/src/instrumentation.dart]: getters should not have hidden O(N) algorithms. :-)
Should probably be a method.

Comment (eseidelGoogle) [inline packages/flutter_test/lib/src/instrumentation.dart]: I just moved it. :p  Where is our memoize decorator?

Comment (Hixie) [inline packages/flutter_test/lib/src/instrumentation.dart]: never mind, i see this is just code from before

**Comment by Hixie** [comment]
LGTM assuming you fix the indentation.

**Pull request merged**